src/sender.c

```diff
@@ -277,25 +321,24 @@ int send_data(int sender_fd, char *fname, char *pub_key, size_t pub_len,
     status =
         retry_send(sender_fd, payload, GET_PAYLOAD_PACKET_LEN(ctext_len), 0);
     free(payload);
-    free(ctext);
     if (status == -1) {
       error(sender_fd, "Send data failed");
       return -1;
     }
+    accumulated_sz += ctext_len;
 
     if (finish_send) break;
   }
 
-  SHA256_Final(hash, &sha256);
-
-  for (int i = 0; i < SHA256_DIGEST_LENGTH; i++) {
-    sprintf(sha256_str + (i * 2), "%02x", hash[i]);
-  }
-
   // End of data transfer
   debug(sender_fd, "Finish sending file");
-  fclose(src_file);
+  return 0;
+}
 
+int compare_check_sum(int sender_fd, char sha256_str[65]) {
```

Comment:
```suggestion
int send_checksum(int sender_fd, char sha256_str[65]) {
```
Since this function do nothing related to compare and checksum should be a single word. I think it would be better to rename it as `send_checksum`